tests/committable.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import infra.e2e_args
+import infra.network
+import infra.proc
+import time
+
+from loguru import logger as LOG
+
+
+def run(args):
+    hosts = ["localhost"] * 5
+
+    with infra.network.network(
+        hosts, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+    ) as network:
+        network.start_and_join(args)
+        primary, backups = network.find_nodes()
+
+        # Suspend three of the backups to prevent commit
+        backups[1].suspend()
+        backups[2].suspend()
+        backups[3].stop()
+
+        txs = []
+        # Run some transactions that can't be committed
+        with primary.client("user0") as uc:
+            for i in range(10):
+                txs.append(
+                    uc.post("/app/log/private", {"id": 100 + i, "msg": "Hello world"})
+                )
+
+        # Wait for a signature to ensure those transactions are committable
```

Comment:
This kind of sleep is something we do in several places. It now potentially doesn't do what was intended - if we're in the fancy election state and Raft is not returning a `signable` index, we're simply skipping signatures and not producing them at regular intervals.

This is an argument in favour of continuing to produce signatures (making a max-lengthed block of transactions _committable_) without any global commit claim in them, _but_ that's uglier for ledger parsing. I'm happy to keep the current behaviour (signatures are sometimes skipped) for this PR and discuss alternative approaches offline/in future.

Reply:
Yes that's true although that will not cause the test to silently pass.

The issue here is that we do not have a way to observe what's committable, the real condition would be, the last of those tx is committable on backup[1]. Even if we shortened the sigx-tx-interval, we wouldn't know that a signature was produced and replicated to backup[1].